content/docs/ops/traffic-management/troubleshooting/index.md

```diff
@@ -62,6 +62,91 @@ to propagate to all the sidecars.  With large deployments the
 propagation will take longer and there may be a lag time on the
 order of seconds.
 
+## Destination rule policy not activated
+
+Although destination rules are associated with a particular destination host,
+the activation of subset-specific policies depends on route rule evaluation.
+
+The first step in Envoy's service routing process is to evaluate the route rules in
+a virtual service corresponding to the requested host, if there are any,
+to determine if a subset (that is, specific version) of the destination is being routed to.
+If so, the policies corresponding to the subset, if any, are then evaluated.
+Consequently, policies that are defined for specific subsets are only applied if
+the corresponding subset is explicitly routed to.
+
+For example, consider the following destination rule as the one and only configuration defined for the
+*reviews* service, that is, there are no route rules in a corresponding virtual service definition:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: reviews
+spec:
+  host: reviews
+  subsets:
+  - name: v1
+    labels:
+      version: v1
+    trafficPolicy:
+      connectionPool:
+        tcp:
+          maxConnections: 100
+{{< /text >}}
+
+Since there is no specific route rule defined for the *reviews*
+service, default round-robin routing behavior will apply, which will
+presumably call "v1" instances on occasion, maybe even always if "v1" is
+the only running version. Nevertheless, the above policy will never be
+invoked since the default routing is done at a lower level. The rule
+evaluation engine will be unaware of the final destination and therefore
+unable to match the subset policy to the request.
+
+You can fix the above example in one of two ways. You can either move the
+traffic policy up a level in the destination rule to make it apply to any subset:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: reviews
+spec:
+  host: reviews
+  trafficPolicy:
+    connectionPool:
+      tcp:
+        maxConnections: 100
+  subsets:
+  - name: v1
+    labels:
+      version: v1
+{{< /text >}}
+
+Or, better yet, define proper route rules for the service in the virtual service definition.
+For example, add a simple route rule for "reviews:v1":
+
+{{< text yaml >}}
```

Comment:
This virtual service is a perfect example of what I meant regarding the name of the destination rule. The same value, `reviews` is used in this example for the name of the virtual service, the service, and the host name for destination rule. It would be incredibly helpful if we used different names for each in our examples. We can make a separate recommendation for production environments where I think there might be some ease-of-use value in using the same name for everything. Please change the name of the virtual service, the destination rule, and the corresponding values of `host` and `hosts` for clarity.

Reply:
First of all, this example is from our Bookinfo sample application, so using a different naming convention is not an option. Second, we had a discussion about naming conventions for virtual service and destination rules quite a while ago and decided this is a nice simple pattern, so we used it in Bookinfo. Third, this is the operations guide, not the Istio 101 concepts, so readers are already using Istio and should not be so easily confused.